test/hotspot/jtreg/serviceability/sa/ClhsdbSymbol.java

```diff
@@ -51,56 +51,50 @@ public static void main(String[] args) throws Exception {
             theApp = LingeredApp.startApp();
             System.out.println("Started LingeredApp with pid " + theApp.getPid());
 
-            //Use command "class java.lang.Thread" to get the address of the InstanceKlass for java.lang.Thread
+            // Use command "class java.lang.Thread" to get the address of the InstanceKlass for java.lang.Thread
             cmdStr = "class java.lang.Thread";
             cmds = List.of(cmdStr);
             expStrMap = new HashMap<>();
             expStrMap.put(cmdStr, List.of("java/lang/Thread"));
             String classOutput = test.run(theApp.getPid(), cmds, expStrMap, null);
-            String threadAddress = null;
-            String[] parts = classOutput.split("\n");
 
-            //extract thread address from the output line similar to "java/lang/Thread @0x000000080001d940"
-            for (String part : parts) {
-                if (part.startsWith("java/lang/Thread")) {
-                    String[] addresses = part.split(" @");
-                    threadAddress = addresses[1];
-                    break;
-                }
-            }
+            // Extract the thread InstanceKlass address from the output, which looks similar to the following:
+            //   java/lang/Thread @0x000000080001d940
+            String threadAddress = classOutput.lines().filter(part -> part.startsWith("java/lang/Thread"))
+                                                      .map(part -> part.split(" @"))
+                                                      .findFirst()
+                                                      .map(addresses -> addresses[1])
+                                                      .orElse(null);
```

Comment:
It should just call `get()` because `NoSuchElementException` would be called when we cannot find out Thread class from stdout.
If you want to throw `RuntimeException` like next step, you should use `orElseThrow()`.